### ISSUE TITLE
JAV Library Scraper Broke

### ISSUE THREAD

**Issue by Zero177768** [adultmm/AdultMediaManager#112]
JavLibrary Scraper is throwing a 403 error.
If I manually run it will find the info but can not access it to download.
See screen shots.
![Screenshot 2024-05-13 195807](https://github.com/adultmm/AdultMediaManager/assets/156572347/54af4ec0-7f64-4015-9feb-c76ec69ccddc)
![Screenshot 2024-05-13 195918](https://github.com/adultmm/AdultMediaManager/assets/156572347/3b2be8dd-c0f0-437a-bd9a-699a042281c9)
![Screenshot 2024-05-13 195957](https://github.com/adultmm/AdultMediaManager/assets/156572347/88620582-5a33-4014-b1c9-45e808e3b71f)
![Screenshot 2024-05-13 200037](https://github.com/adultmm/AdultMediaManager/assets/156572347/20555d53-ebd7-4e21-bb6c-003f6951eb5f)

Thanks :)
Troy

**Comment by Zero177768** [comment]
I think this is a CloudFlare issue

**Comment by Zero177768** [comment]
![Untitled](https://github.com/adultmm/AdultMediaManager/assets/156572347/fc3a7fa3-5c09-4ac8-9dd8-2c420b193aba)

**Comment by Zero177768** [comment]
But This one works After I pass Cloud 
![Untitled2](https://github.com/adultmm/AdultMediaManager/assets/156572347/68d10b63-1607-4c4c-aad2-6b7163444834)
Flare

**Comment by Zero177768** [comment]
Is there an Addon for Ember that can pass cloud Flare ? 
Or is this not possible?

**Comment by Zero177768** [comment]
I can get Plex to pull Metadata using the instructions below, But Plex does not add the posters and metadata to the folder where the file is located only ember will do that.


Here are the instructions for plex to pull Metadata:

https://github.com/Xavier-Lam/JAV.bundle?tab=readme-ov-file


JAVLibrary
Recently, [JAVLibrary](https://javlibrary.com/) reenabled cloudflare challenge. To bypass it, you have to follow these steps:

1) Visit [JAVLibrary](https://javlibrary.com/) on your server and pass the challenge.
2) After you can see content on JAVLibrary

        2.1)open developer tools of your browser by pressing F12,

        2.2)refresh the page

        2.3)choose Network tab and select Doc

        2.4)choose the first request, copy the User-Agent header in your request and cf_clearance cookie value to the 
              configuration in your plex agent Settings

Note:

The source IP address you visit JAVLibrary should be identical to your plex server outbound IP address, if you are using a proxy, the address would be the outbound IP address of your proxy;
The cf_clearance value may change frequently, especially when you are using a proxy, every time your IP changes, the cf_clearance changes;
Every time you update your browser, the User-Agent would change.
Or simply, you can use this project: [CF-Clearance-Scraper](https://github.com/Xewdy444/CF-Clearance-Scraper/tree/main) to get cf_clearance cookie.

**Comment by adultmm** [comment]
Yes, this scraper is broken because of the CloudFlare challenge. Unfortunately, I haven't found any solution for this type of challenge yet, and [FlareSolverr](https://github.com/FlareSolverr/FlareSolverr) doesn't work with it either.

Every idea or solution for this problem is very welcome.

**Comment by adultmm** [comment]
The [CF-Clearance-Scraper](https://github.com/Xewdy444/CF-Clearance-Scraper/tree/main) is not working either. 

>python main.py -v -f cookies.json https://www.javlibrary.com/en/?v=javme5tj34
[19:14:24] [INFO] Launching headless browser...
[19:14:26] [INFO] Going to https://www.javlibrary.com/en/?v=javme5tj34...
[19:14:27] [INFO] Solving Cloudflare challenge [Managed]...
[19:14:58] [ERROR] Failed to retrieve a Cloudflare clearance cookie.

**Comment by adultmm** [comment]
Jav.Guru scraper has been implemented in [v1.0.0-beta2](https://github.com/adultmm/AdultMediaManager/releases/tag/v1.0.0-beta2). 

There isn't runtime information and fewer images, but it works as far as I could test. Give it a try and let me know if you find any problems.
